airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -116,10 +118,18 @@ public Optional<Long> createResetConnectionJob(final DestinationConnection desti
             workerResourceRequirements))
         .withResetSourceConfiguration(new ResetSourceConfiguration().withStreamsToReset(streamsToReset));
 
+    getCurrentConnectionState(standardSync.getConnectionId()).ifPresent(resetConnectionConfig::withState);
```

Comment:
Seems like this could be abstracted to a method that takes the function to call if present instead of duplicating between read and reset.